### PR TITLE
chore(install): use generated types for installation recipes

### PR DESCRIPTION
**Pull request by sanderblue**
**Checklist**
- [ ] ~Remove recipe_file.go? (yes, please 🦙 )~  Can't quite do this just yet. Next PR
- [x] Need a custom `UnmashalYAML()` method for `OpenInstallationRecipe` due to the `Install` field
- [x] How to handle the `Vars` thing for recipes variable injection
    - update `AddVar()` to use `InputVars` since it that field already exists on `OpenInstallationRecipe`
- [x] Remove `file` from the NerdGraph query
- [x] Remove usage of `OpenInstallationRecipe.File` (deprecated and no longer needed)
- [x] Add note to the custom unmarshal method for the recipe struct for added context

### CODE REVIEW

**Comment by zlesnr** [comment]
Don't forget to integrate the `vars` from the `OpenInstallRecipe`.

**Comment by zlesnr** [comment]
Also don't forget that `Install` on the `OpenInstallRecipe` is a `string` currently, but should be a `map[string]interface{}` or some other solution there so we have the go-task bits.

File: internal/install/recipe_installer_targeted.go [hunk 1]
```diff
@@ -125,33 +124,33 @@ func (i *RecipeInstaller) targetedInstall(ctx context.Context, m *types.Discover
...
 }
 
-func finalizeRecipe(f *recipes.RecipeFile) (*types.Recipe, error) {
-	r, err := f.ToRecipe()
-	if err != nil {
-		return nil, fmt.Errorf("could not finalize recipe %s: %s", f.Name, err)
-	}
-	return r, nil
-}
+// func finalizeRecipe(f *types.OpenInstallationRecipe) (*types.OpenInstallationRecipe, error) {
```
[the remark this answers is not in the document]
Comment (sanderblue) [reply internal/install/recipe_installer_targeted.go:145]: Yep, thanks for catching that. Needed another set of eyes.

File: internal/install/recipe_installer.go [hunk 2]
```diff
@@ -167,14 +167,16 @@ func (i *RecipeInstaller) assertDiscoveryValid(ctx context.Context, m *types.Dis
...
-func (i *RecipeInstaller) installRecipes(ctx context.Context, m *types.DiscoveryManifest, recipes []types.Recipe) error {
+func (i *RecipeInstaller) installRecipes(ctx context.Context, m *types.DiscoveryManifest, recipes []types.OpenInstallationRecipe) error {
 	log.WithFields(log.Fields{
 		"recipe_count": len(recipes),
 	}).Debug("installing recipes")
 
 	for _, r := range recipes {
 		var err error
 
+		r.SetRecipeVar("DISCOVERED_LOG_FILES", "/testing/log/files/path/string")
```
[the remark this answers is not in the document]
Comment (sanderblue) [reply internal/install/recipe_installer.go:178]: Indeed, will remove. Gracias 😎

**Pull request merged**